### PR TITLE
Upgrade Calagator to Rails 5

### PR DESCRIPTION
**Pull request by aeschright**
TODO:
- [x] Rename dummy app
- [x] Fix SQL search queries
- [x] Fix tag assignment errors in failing tests
- [x] Remove master/slave terminology from duplicate checking
- [x] Get all tests to pass
- [x] Update docs

### CODE REVIEW

**Comment by aeschright** [comment]
Down to five failing tests!

**Comment by aeschright** [comment]
Looks like that syntax applies to choosing a version for the Calagator gem, not Rails. It really seems like there ought to be some command line setting or environment setting I could use, but I'm having a hard time finding anything relevant in the docs.

**Comment by aeschright** [comment]
I was considering updating our forms to use the recaptcha v3 workflow, but the gem will allow us to continue with v2 and the changes started to feel out of scope for this PR. Issue #554 is still open for discussion of recaptcha and other spam-control options.

File: spec/test_app/db/migrate/20211013183057_add_missing_unique_indices.acts_as_taggable_on_engine.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# This migration comes from acts_as_taggable_on_engine (originally 2)
```
Comment (reidab) [inline spec/test_app/db/migrate/20211013183057_add_missing_unique_indices.acts_as_taggable_on_engine.rb:1]: Do we need these migrations here as well as in the engine? The copy in the engine should get run in the host app.

**Review by reidab — APPROVED** [review]

File: app/models/calagator/event.rb [hunk 2]
```diff
@@ -30,42 +30,45 @@
...
-    validates :title, :description, :url, blacklist: true
+    validates :title, :description, :url, denylist: true
     validates :start_time, :end_time, sequential: true
     validates :title, :start_time, presence: true
     validates :url,
               format: { with: %r{\Ahttps?://(\w+:?\w*@)?(\S+)(:[0-9]+)?(/|/([\w#!:.?+=&%@!\-/]))?\Z},
                         allow_blank: true }
 
-    before_destroy { !locked } # prevent locked events from being destroyed
+    before_destroy :check_if_locked_before_destroy # prevent locked events from being destroyed
```
[the remark this answers is not in the document]
Comment (aeschright) [reply app/models/calagator/event.rb:64]: Leaving this as is for now

File: spec/test_app/db/migrate/20211013183057_add_missing_unique_indices.acts_as_taggable_on_engine.rb (hunk 1 above)
Comment (aeschright) [reply spec/test_app/db/migrate/20211013183057_add_missing_unique_indices.acts_as_taggable_on_engine.rb]: Removed 👍🏻

**Pull request merged**